projects/gmx/index.js

```diff
@@ -2,83 +2,55 @@ const sdk = require('@defillama/sdk');
+const avalancheGMX = '0x62edc0692BD897D2295872a9FFCac5425011c661'
 
-const arbitrumTVL = async (timestamp, block, chainBlocks) => {
-    let balances = {};
-    
-    balances['weth'] = await balanceOf(WETH, VAULT, chainBlocks);
-    balances['wrapped-bitcoin'] = await balanceOf(WBTC, VAULT, chainBlocks);
-    balances['usd-coin'] = await balanceOf(USDC, VAULT, chainBlocks);
-    balances['tether'] = await balanceOf(USDT, VAULT, chainBlocks);
-    balances['chainlink'] = await balanceOf(LINK, VAULT, chainBlocks);
-    balances['uniswap'] = await balanceOf(UNI, VAULT, chainBlocks);
-    balances['magic-internet-money'] = await balanceOf(MIM, VAULT, chainBlocks);
-    balances['frax'] = await balanceOf(FRAX, VAULT, chainBlocks);
-    balances['dai'] = await balanceOf(DAI, VAULT, chainBlocks);
-    //balances['aave'] = await balanceOf(AAVE, VAULT, chainBlocks); Arbitrum address TBD
-    balances['spell-token'] = await balanceOf(SPELL, VAULT, chainBlocks);
-    balances['sushi'] = await balanceOf(SUSHI, VAULT, chainBlocks);
-    
-    return balances;
-}
-
-const bscTVL = async (timestamp, block, chainBlocks) =>{
-    const balances = {}
-    const allTokens = (await axios.get(apiEndpoint)).data
-    const tokenBalances = await sdk.api.abi.multiCall({
-        calls: allTokens.map(token=>({
-            target: token.id,
-            params: [pool]
-        })),
-        abi: 'erc20:balanceOf',
-        chain: 'bsc',
-        block: chainBlocks.bsc
-    })
-    sdk.util.sumMultiBalanceOf(balances, tokenBalances, true, d=>`bsc:${d}`)
-    return balances
+const arbitrumTVL = async (timestamp, block, chainBlocks) =>{
+  const balances = {}
+  const allTokens = (await axios.get(arbitrumApiEndpoint)).data
+  const tokenBalances = await sdk.api.abi.multiCall({
+      calls: allTokens.map(token=>({
+          target: token.id,
+          params: [arbitrumVault]
+      })),
+      abi: 'erc20:balanceOf',
+      chain: 'arbitrum',
+      block: chainBlocks.arbitrum
+  })
+  sdk.util.sumMultiBalanceOf(balances, tokenBalances, true, d=>`arbitrum:${d}`)
+  return balances
 }
 
-async function balanceOf(pTarget, pParam, pChainBlocks) {
-    let decimals = await sdk.api.abi.call({
-        target: pTarget,
-        abi: 'erc20:decimals',
-        block: pChainBlocks.arbitrum,
-        chain: 'arbitrum'
-    });
-    let result = await sdk.api.abi.call({
-      target: pTarget,
-      params: pParam,
+const avalancheTVL = async (timestamp, block, chainBlocks) =>{
+  const balances = {}
+  const allTokens = (await axios.get(avalancheApiEndpoint)).data
+  const tokenBalances = await sdk.api.abi.multiCall({
+      calls: allTokens.map(token=>({
+          target: token.id,
+          params: [avalancheVault]
+      })),
       abi: 'erc20:balanceOf',
-      block: pChainBlocks.arbitrum,
-      chain: 'arbitrum'
-    });
-    return result.output*Math.pow(10, -decimals.output);
+      chain: 'avax',
+      block: chainBlocks.arbitrum
```

Comment:
Just saw that avalanche tvl calculation seemed to work even though i specified block as chainBlocks.arbitrum? Can someone explain briefly, or should I change it to chainBlocks.avax?

Reply:
you probably want chainBlocks.avax if you're reading data from the avax chain. chainBlocks.arbitrum will go through without errors because avax has a higher block number than arbitrum (at the moment at least). So you're basically reading old data on avax by accident

Reply:
alright i will fix thanks for the explaination